src/dto/proposals-response.js

```diff
@@ -16,20 +16,22 @@
  */
 
 // @flow
-import ProposalDTO from './proposal'
+import { parseProposalDTO } from './proposal'
+import type { ProposalDTO } from './proposal'
+import { validate } from '../validation'
 
 type ResponseMapped = {
   proposals: Array<Object>
 }
 
-class ProposalsResponseDTO {
-  proposals: ?Array<ProposalDTO>
+type ProposalsResponseDTO = {
+  proposals: Array<ProposalDTO>
+}
 
-  constructor (responseData: ResponseMapped) {
-    if (typeof responseData.proposals !== 'undefined') {
-      this.proposals = responseData.proposals.map((proposal) => new ProposalDTO(proposal))
-    }
-  }
+function parseProposalsResponseDTO (responseData: ResponseMapped): ProposalsResponseDTO {
+  validate('ProposalResponseDTO', responseData, { name: 'proposals', type: 'array' })
```

Comment:
!! `responseData : ResponseMapped` param indicates that this function expects to get `ResponseMapped` data, but it doesn't need to - it validates that this param has `proposals` array, so it should receive `mixed`/`any`/`Object`

Reply:
Will approve once this is fixed:)